src/couch_replicator_manager.erl

```diff
@@ -410,10 +438,30 @@ rescan(#state{scan_pid = nil} = State) ->
         true ->
             State;
         false ->
+            % Clean up existing change feed pids.
+            % They have to be removed from the list of
+            % pids otherwise resume_scan will never make progress
+            % (it notices an existing pid and assumes there is
+            % a change feed process running, bump rescan flag
+            % and moves on).
+            NewPids = lists:filtermap(
+                        fun({rep_start, Pid}) ->
+                                % Keep replications which are starting
+                                {true, {rep_start, Pid}};
+                           ({_DbName, Pid}) ->
+                                % Kill and throw away change feeds
+                                catch unlink(Pid),
```

Comment:
If we unlink doesn't that mean we don't hear about its death so we'd either have to retrigger all necessary rescans here?